### PR TITLE
Replace Sass variables with custom properties on `_font-weights.pcss`

### PR DESCRIPTION
**Pull request by luixxiul**
For https://github.com/vector-im/element-web/issues/21656 and https://github.com/vector-im/element-web/issues/25073

This PR intends to replace existing Sass variables on `_font-weights.pcss` and declarations with hard-coded value (`font-weight: 600`) with CSS custom properties, based on the [style guide](https://github.com/vector-im/element-web/blob/1ebe574ab1bc3dd55489dfd0ca5ff02e918d976e/code_style.md?plain=1#L288).

Currently there are two custom properties on `_font-weights.pcss` (`--font-normal` and `--font-semi-bold`). Custom properties for the other weight values are yet to be implemented.

The PR is based on the assumption that `font-weight: 600` is always equal to `font-weight: var(--font-semi-bold)`. Please correct me if I am wrong.

type: task

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by richvdh — APPROVED** [review]
great, thanks!

**Comment by luixxiul** [comment]
Fixing the conflict..

**Review by richvdh — CHANGES_REQUESTED** [review]
looks like there's still a reference to `$font-semi-bold` somewhere which is breaking CI

File: res/css/structures/_SpaceRoomView.pcss [hunk 1]
```diff
@@ -27,12 +27,10 @@ $SpaceRoomViewInnerWidth: 428px;
     font-weight: $font-semi-bold;
```
Comment (richvdh) [inline res/css/structures/_SpaceRoomView.pcss]: here, in fact

Comment (luixxiul) [reply res/css/structures/_SpaceRoomView.pcss]: Right, addressed with https://github.com/matrix-org/matrix-react-sdk/pull/10554/commits/9e30d1b96ca8d65835a8d43cfa67d82fe24bdd07.

**Review by richvdh — APPROVED** [review]

**Pull request merged**